core/src/main/scala/spinal/core/sim/package.scala

```diff
@@ -356,50 +358,49 @@ package object sim {
     }
   }
 
-  /**
-    * Add implicit function to Bits
-    */
-  implicit class SimBitsPimper(bt: Bits) {
+  protected abstract class RandomizableBitVector(bt: BitVector, longLimit: Int) {
+    protected val width = bt.getWidth
 
-    def randomize(): Unit = {
-      val width = bt.getWidth
-      if(width < 64){
-        bt #= Random.nextLong() & ((1l << width) - 1)
-      }else {
-        bt #= BigInt(width, Random)
+    def randomize(): BigInt = {
+      if (width < longLimit) {
+        val l = randomizedLong()
+        bt #= l
+        l
+      } else {
+        val bi = randomizedBigInt()
+        bt #= bi
+        bi
       }
     }
 
+    def randomizedBigInt(): BigInt
+
+    def randomizedLong(): Long
+
+    def randomizedInt(): Int
```

Comment:
What about providing a default implementation of all functions as for unsigned, and then only `SInt` would override by substraction?

```scala
override def randomizedBigInt(): BigInt = {
  super.randomizedBigInt() - (BigInt(1) << width)
}
```

Also, isn't it possible to just have the base class `implicit`, not `abstract` (is the word for it "concrete"? "actual"? "instantiable"? "instantiatable"?) and public, and having another implicit class only for `SInt`?